# Consulta Amigable MEF: Análisis Presupuestal a nivel de Gobiernos Locales

_____________________
```
Proyecto   : WS CAMEF
Script     : d_analysis.ipynb
Descripción: Análisis de data y reporte.
Fecha      : 2025-02-15
Versión    : 1.0 (draft)
Autor      : Alex Evanan & Mauricio Beuzeville

H.Revisión:
    - [2025-02-07] v1.0: Versión incial.
    - [2025-03-02] V1.1: Gráficos y tablas
```
_____________________

## Librerías

In [48]:
# Librerias
import pandas as pd
import numpy as np
import os

import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import plotly.graph_objects as go
import plotly.figure_factory as ff
from matplotlib.ticker import StrMethodFormatter


## Carga de Datos

In [49]:
# Directorio de trabajo
dir_path          = os.getcwd()
dir_path_data_pro = os.path.join(os.path.dirname(dir_path), "01_data/02_processed")
dir_path_results  = os.path.join(os.path.dirname(dir_path), "04_results/01_graphs")

# Archivos
file_muni = "PROCESADO_EJECUCION_GASTO_GL_X_MUNICIPALIDADES.xlsx"

# Carga de datos
df = pd.read_excel(
    os.path.join(dir_path_data_pro, file_muni), dtype={col: str for col in range(1, 8)}
)
            
# Información del dataframe
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20674 entries, 0 to 20673
Data columns (total 16 columns):
 #   Column                          Non-Null Count  Dtype  
---  ------                          --------------  -----  
 0   Año                             20674 non-null  int64  
 1   UBI_DPTO                        20674 non-null  object 
 2   Departamento                    20674 non-null  object 
 3   UBI_PROV                        20674 non-null  object 
 4   Provincia                       20674 non-null  object 
 5   UBI_DIST                        20674 non-null  object 
 6   COD_SIAF                        20674 non-null  object 
 7   Municipalidad                   20674 non-null  object 
 8   PIA                             20674 non-null  int64  
 9   PIM                             20674 non-null  int64  
 10  Certificación                   20674 non-null  int64  
 11  Compromiso Anual                20674 non-null  int64  
 12  Atención de Compromiso Mensual  

## Análisis

In [50]:
# Procesamiento de datos

# Ajuste de etiquetas
df = df.replace('PROVINCIA CONSTITUCIONAL DEL CALLAO',"CALLAO")

df.head()


Año UBI_DPTO Departamento UBI_PROV    Provincia UBI_DIST COD_SIAF  \
0  2015       01     AMAZONAS     0101  CHACHAPOYAS   010101   300001   
1  2015       01     AMAZONAS     0101  CHACHAPOYAS   010102   300002   
2  2015       01     AMAZONAS     0101  CHACHAPOYAS   010103   300003   
3  2015       01     AMAZONAS     0101  CHACHAPOYAS   010104   300004   
4  2015       01     AMAZONAS     0101  CHACHAPOYAS   010105   300005   

                             Municipalidad       PIA       PIM  Certificación  \
0  MUNICIPALIDAD PROVINCIAL DE CHACHAPOYAS  11099547  72422840       67817314   
1      MUNICIPALIDAD DISTRITAL DE ASUNCION    370554   1089344         387442   
2        MUNICIPALIDAD DISTRITAL DE BALSAS    403348    598968         581088   
3         MUNICIPALIDAD DISTRITAL DE CHETO    394983   1899721        1859456   
4     MUNICIPALIDAD DISTRITAL DE CHILIQUIN    401805   1252502         533946   

   Compromiso Anual  Atención de Compromiso Mensual  Devengado    Girado  \
0          45827779                        45827779   45820914  45811893   
1            395018                          395018     395018    392618   
2            581088                          581088     581064    581064   
3            694472                          694472     694472    694472   
4            526948                          465044     465044    465043   

   Avance %  
0      63.3  
1      36.3  
2      97.0  
3      36.6  
4      37.1

### PIM, Devengado y Avance

In [51]:
# Grafico PIM, Devengado y Avance

# Data para el gráfico
df_gph = (
    df.loc[df["Año"] <= 2025]
    .groupby("Año", as_index=False)
    .agg({
        "PIM": lambda x: x.sum() / 1_000_000,
        "Devengado": lambda x: x.sum() / 1_000_000,
        "Avance %": "mean"
    })
).copy()

df_gph.round(2)

Año       PIM  Devengado  Avance %
0   2015  28902.33   21098.69     74.39
1   2016  31373.91   22980.62     74.28
2   2017  34902.69   24134.20     73.25
3   2018  37850.76   27091.54     75.22
4   2019  35634.44   24668.12     72.48
5   2020  42727.58   27057.78     67.54
6   2021  45000.85   32231.19     74.68
7   2022  52860.45   37737.23     76.88
8   2023  49714.72   36100.96     75.79
9   2024  51766.26   39369.36     71.56
10  2025  40665.84    4435.95      8.87

En el período **2015-2024**, el avance en la **ejecución presupuestal** de los gobiernos locales alcanzó su punto más bajo en **2020**, con un **68%**. Por otro lado, el **2022** registró el mayor avance, con un **77%**, evidenciando una mejora en la gestión y ejecución del presupuesto. En promedio, la ejecución presupuestal a nivel de municipalidades es de **73%**. Es decir 27 de cada 100 soles son devueltos al MEF.

![](../04_results/01_graphs/gph_03a.png)
<!-- <img src="../04_results/01_graphs/gph_03a.png" alt="Gráfico" width="600" height="auto"> -->


### Participación PIM Departamental

In [52]:
# Grafico PIM por departamento v2

# Data para el gráfico
df_gph = df.loc[df["Año"] <= 2025].pivot_table(
    index  =["Departamento"],
    columns=["Año"],
    values =["PIM"],
    aggfunc={
        "PIM": lambda x: x.sum() / 1_000_000,
        },
    # fill_value=0,
    margins=False,
    margins_name="Total",
    dropna=True,
    observed=True,
).copy()

# Eliminar el nivel 0
df_gph.columns = df_gph.columns.droplevel(0)

# Columna promedio
df_gph["Prom."] = df_gph.mean(axis=1)

# ordenar segun promedio
df_gph = df_gph.sort_values(by="Prom.", ascending=False)

# Transformar en porcentajes
df_gph = round(df_gph.div(df_gph.sum(axis=0), axis=1) * 100,2)

df_gph

Año             2015   2016   2017   2018   2019   2020   2021   2022   2023  \
Departamento                                                                   
LIMA           20.68  20.66  21.73  22.64  20.50  18.23  17.92  15.44  15.83   
CUSCO           9.23   8.16   7.71   8.65   8.71   7.36   8.22  10.47  12.06   
ANCASH          4.45   4.50   5.19   6.34   6.95   7.14   8.17   9.68   9.27   
PIURA           5.83   5.52   5.27   5.69   6.87   6.94   7.03   6.63   7.32   
AREQUIPA        5.62   5.55   5.34   6.09   5.84   6.21   6.69   7.33   6.45   
LA LIBERTAD     5.32   5.40   5.53   5.99   6.99   6.65   6.25   5.46   5.20   
CAJAMARCA       5.80   6.24   6.16   5.15   5.43   6.11   6.22   5.62   5.22   
PUNO            4.93   4.78   4.47   3.60   3.41   4.06   3.62   3.53   3.44   
JUNIN           3.86   3.96   3.88   3.79   3.62   3.76   3.40   3.28   3.32   
AYACUCHO        3.44   3.49   3.71   3.51   3.68   4.18   3.81   3.15   2.96   
ICA             2.61   2.36   2.16   2.28   2.27   2.34   2.74   3.92   4.01   
APURIMAC        3.14   3.69   3.58   3.25   2.54   2.59   2.76   2.83   2.35   
HUANUCO         2.99   3.23   3.60   3.08   2.65   2.74   2.50   2.30   2.22   
LORETO          2.58   2.86   2.65   2.47   2.58   2.65   2.36   2.21   2.44   
SAN MARTIN      2.50   2.62   3.01   2.72   2.85   3.02   2.66   2.17   2.12   
LAMBAYEQUE      2.62   2.57   2.49   2.27   2.26   2.58   2.63   2.36   2.37   
HUANCAVELICA    2.34   2.51   2.68   2.16   2.21   2.52   2.44   1.94   1.85   
CALLAO          2.27   2.38   2.18   2.05   2.14   2.17   2.03   1.97   2.13   
TACNA           2.01   1.79   1.61   1.52   1.47   1.61   1.99   2.64   2.43   
MOQUEGUA        1.89   1.82   1.34   1.31   1.42   1.42   1.64   2.00   1.83   
AMAZONAS        1.62   1.50   1.70   1.56   1.53   1.70   1.48   1.34   1.55   
UCAYALI         1.47   1.65   1.52   1.51   1.55   1.40   1.26   1.44   1.55   
PASCO           1.48   1.44   1.37   1.21   1.28   1.22   0.98   1.08   0.96   
TUMBES          1.00   0.94   0.75   0.78   0.78   0.87   0.76   0.78   0.76   
MADRE DE DIOS   0.32   0.39   0.39   0.35   0.48   0.51   0.43   0.44   0.38   

Año             2024   2025  Prom.  
Departamento                        
LIMA           18.88  21.03  19.08  
CUSCO          10.17   9.96   9.30  
ANCASH          8.71   7.63   7.41  
PIURA           6.35   5.88   6.38  
AREQUIPA        6.05   6.22   6.21  
LA LIBERTAD     4.77   4.76   5.64  
CAJAMARCA       4.82   4.85   5.56  
PUNO            3.25   2.98   3.74  
JUNIN           3.30   3.27   3.54  
AYACUCHO        2.94   2.57   3.37  
ICA             3.87   4.11   3.08  
APURIMAC        2.55   2.51   2.84  
HUANUCO         2.43   2.45   2.68  
LORETO          3.01   3.47   2.65  
SAN MARTIN      2.48   2.21   2.55  
LAMBAYEQUE      2.20   2.36   2.42  
HUANCAVELICA    1.86   1.83   2.18  
CALLAO          2.27   2.44   2.17  
TACNA           2.37   2.32   2.03  
MOQUEGUA        2.19   2.00   1.74  
AMAZONAS        1.79   1.51   1.57  
UCAYALI         1.53   1.58   1.49  
PASCO           1.11   0.95   1.16  
TUMBES          0.74   0.69   0.79  
MADRE DE DIOS   0.36   0.40   0.41

Según la distribución porcentual del **PIM** (a nivel de gobiernos locales) por departamentos (2015-2025), **Lima** lidera con un promedio del **19.08%**, aunque su participación ha disminuido entre 2018-2023. Mientras que **Madre de Dios** y **Tumbes** tienen la menor participación con menos de **1%** en promedio.

![](../04_results/01_graphs/gph_04b.png)

### Relación N° Municipalidades y PIM

In [53]:
# Tabla Num Municipalidades

# Data para el gráfico
df_gph = df.loc[df["Año"] == 2025].pivot_table(
    index  =["Departamento"],
    columns=["Año"],
    values =[
        "Municipalidad",
        # "Municipalidad",
        ],
    aggfunc={
        "Municipalidad": [
            lambda x: x.str.contains("PROVINCIAL|METROPOLITANA").sum(),  # Contar "PROVINCIAL"
            lambda x: x.str.contains("DISTRITAL").sum(),   # Contar "DISTRITAL"
        ]
        },
    # fill_value=0,
    margins=False,
    margins_name="Total",
    dropna=True,
    observed=True,
).copy()

# Eliminar el nivel 0
df_gph.columns = df_gph.columns.droplevel(0)

#Renombrar columnas
df_gph.columns = ["Muni. Provincial", "Muni. Distrital"]

# ordenar segun promedio
df_gph = df_gph.sort_values(by="Muni. Distrital", ascending=False)

# Agregar totales de filas y columnas
df_gph = df_gph.assign(Total=df_gph.sum(axis=1))  # Totales de filas

df_gph

Muni. Provincial  Muni. Distrital  Total
Departamento                                           
LIMA                         10              161    171
ANCASH                       20              146    166
JUNIN                         9              115    124
CAJAMARCA                    13              114    127
AYACUCHO                     11              113    124
CUSCO                        13              103    116
AREQUIPA                      8              101    109
PUNO                         13               97    110
HUANCAVELICA                  7               95    102
APURIMAC                      7               78     85
AMAZONAS                      7               77     84
HUANUCO                      11               73     84
LA LIBERTAD                  12               72     84
SAN MARTIN                   10               68     78
PIURA                         8               57     65
LORETO                        8               45     53
ICA                           5               38     43
LAMBAYEQUE                    3               35     38
PASCO                         3               26     29
TACNA                         4               24     28
MOQUEGUA                      3               18     21
UCAYALI                       4               15     19
TUMBES                        3               10     13
MADRE DE DIOS                 3                8     11
CALLAO                        1                6      7

Si bien **Lima** lidera la asignación del PIM con un promedio del **19.08%**, el gráfico de dispersión muestra que también es el departamento con más distritos. Aunque se observa una **tendencia positiva** entre el **número de distritos** y el **PIM promedio**, esta relación no es proporcional en todos los casos. Esto sugiere que, si bien la cantidad de distritos influye en la asignación del PIM, no es el único factor determinante.

![](../04_results/01_graphs/gph_05a.png)

### PIM por Distritos de Lima Metropolitana

In [54]:
# Gráfico PIM por Distritos de Lima Met.

# Data para el gráfico
df_gph = df.loc[
    (df["Año"] <= 2024)&
    (df["Provincia"]=="LIMA")&
    (df["Municipalidad"].str.contains("DISTRITAL"))
    ].pivot_table(
    index  =["Municipalidad"],
    columns=["Año"],
    values =[
        "PIM",
        # "Avance %",
        ],
    aggfunc={
        "PIM": lambda x: x.sum() / 1_000_000,
        # "Avance %": "mean",
        },
    # fill_value=0,
    margins=False,
    margins_name="Total",
    dropna=True,
    observed=True,
).copy()


# Eliminar el nivel 0
df_gph.columns = df_gph.columns.droplevel(0)

# Columna promedio
df_gph["Prom."] = df_gph.mean(axis=1)

# ordenar segun promedio
df_gph = df_gph.sort_values(by="Prom.", ascending=False)

df_gph

Año                                                       2015        2016  \
Municipalidad                                                                
MUNICIPALIDAD DISTRITAL DE SANTIAGO DE SURCO        241.101572  265.395153   
MUNICIPALIDAD DISTRITAL DE ATE - VITARTE            217.467838  259.955380   
MUNICIPALIDAD DISTRITAL DE SAN ISIDRO               211.389121  238.776481   
MUNICIPALIDAD DISTRITAL DE SAN JUAN DE LURIGANCHO   170.500676  228.024979   
MUNICIPALIDAD DISTRITAL DE MIRAFLORES               193.129622  197.758827   
MUNICIPALIDAD DISTRITAL DE SAN MARTIN DE PORRES     128.088269  135.340928   
MUNICIPALIDAD DISTRITAL DE LA MOLINA                123.322621  155.263349   
MUNICIPALIDAD DISTRITAL DE VILLA EL SALVADOR        115.305158  123.584245   
MUNICIPALIDAD DISTRITAL DE PUENTE PIEDRA            117.044155  125.280270   
MUNICIPALIDAD DISTRITAL DE CHORRILLOS               100.256684  110.930256   
MUNICIPALIDAD DISTRITAL DE SAN BORJA                120.891761  124.971968   
MUNICIPALIDAD DISTRITAL DE COMAS                     89.198333  105.655842   
MUNICIPALIDAD DISTRITAL DE LA VICTORIA              100.201309  118.176331   
MUNICIPALIDAD DISTRITAL DE VILLA MARIA DEL TRIUNFO   83.981615  103.871039   
MUNICIPALIDAD DISTRITAL DE LURIGANCHO (CHOSICA)     109.579550   89.918987   
MUNICIPALIDAD DISTRITAL DE SAN JUAN DE MIRAFLORES    92.952002  108.277883   
MUNICIPALIDAD DISTRITAL DE CARABAYLLO                97.539569   79.149190   
MUNICIPALIDAD DISTRITAL DE LOS OLIVOS               108.890910  118.833538   
MUNICIPALIDAD DISTRITAL DE SAN MIGUEL                86.688007   85.947911   
MUNICIPALIDAD DISTRITAL DE JESUS MARIA               70.479895   69.392923   
MUNICIPALIDAD DISTRITAL DE LURIN                     54.618099   65.616904   
MUNICIPALIDAD DISTRITAL DE SANTA ANITA               70.894422  106.593353   
MUNICIPALIDAD DISTRITAL DE INDEPENDENCIA             50.981713   51.692853   
MUNICIPALIDAD DISTRITAL DE PACHACAMAC                60.533536   72.353690   
MUNICIPALIDAD DISTRITAL DE EL AGUSTINO               56.716138   82.647708   
MUNICIPALIDAD DISTRITAL DE MAGDALENA DEL MAR         47.426193   39.395532   
MUNICIPALIDAD DISTRITAL DE RIMAC                     43.384552   55.060798   
MUNICIPALIDAD DISTRITAL DE SURQUILLO                 44.516259   43.145106   
MUNICIPALIDAD DISTRITAL DE LINCE                     41.249392   41.222444   
MUNICIPALIDAD DISTRITAL DE PUEBLO LIBRE              41.239021   48.084558   
MUNICIPALIDAD DISTRITAL DE BREÑA                     25.299557   49.759374   
MUNICIPALIDAD DISTRITAL DE BARRANCO                  39.335432   32.510548   
MUNICIPALIDAD DISTRITAL DE SAN LUIS                  27.325774   48.048571   
MUNICIPALIDAD DISTRITAL DE CIENEGUILLA               30.750224   28.877412   
MUNICIPALIDAD DISTRITAL DE ANCON                     28.266977   29.679185   
MUNICIPALIDAD DISTRITAL DE CHACLACAYO                22.811933   20.483273   
MUNICIPALIDAD DISTRITAL DE PUNTA HERMOSA             17.219007   16.943847   
MUNICIPALIDAD DISTRITAL DE PUCUSANA                  13.058932   13.428196   
MUNICIPALIDAD DISTRITAL DE PUNTA NEGRA               10.341944    9.586423   
MUNICIPALIDAD DISTRITAL DE SAN BARTOLO               18.539761   12.736992   
MUNICIPALIDAD DISTRITAL DE SANTA ROSA                 9.346189    8.284245   
MUNICIPALIDAD DISTRITAL DE SANTA MARIA DEL MAR        7.387060    6.696815   

Año                                                       2017        2018  \
Municipalidad                                                                
MUNICIPALIDAD DISTRITAL DE SANTIAGO DE SURCO        273.491925  298.191681   
MUNICIPALIDAD DISTRITAL DE ATE - VITARTE            296.205306  314.230074   
MUNICIPALIDAD DISTRITAL DE SAN ISIDRO               235.450881  252.874501   
MUNICIPALIDAD DISTRITAL DE SAN JUAN DE LURIGANCHO   227.348237  221.157253   
MUNICIPALIDAD DISTRITAL DE MIRAFLORES               204.730412  227.623191   
MUNICIPALIDAD DISTRIT

Distritos como **Santiago de Surco**, **Ate-Vitarte** y **San Isidro** tienen los PIM más altos, con promedios de **318**, **280** y **275** millones de soles, respectivamente. En contraste, distritos como **Santa María del Mar**, **Santa Rosa** y **San Bartolo** registran los PIM más bajos, con promedios inferiores a **14** millones de soles.

![](../04_results/01_graphs/gph_06a.png)

Al aplicar el método de **agrupamiento jerárquico** (Ward Jr), se identifican con mayor claridad **grupos de distritos** con asignaciones presupuestarias similares a lo largo del tiempo. Algunos presentan niveles persistentemente elevados, mientras que otros comparten características de menor magnitud. La estructura del **dendrograma** (eje izquierdo) sugiere al menos **7 grupos** bien definidos y revela la proximidad entre ellos. Curiosamente, dentro de estos grupos se observa una **tendencia** a la aglomeración de algunos distritos según su **ubicación geográfica** como **Lima centro, norte y sur**.

![](../04_results/01_graphs/gph_06b.png)

### Ejecuación por Distritos de Lima Metropolitana

In [55]:
# Grafico Ejecución Distritos de Lima Meretropolitana V2

# Data para el gráfico
df_gph = df.loc[
    (df["Año"] <= 2024)&
    (df["Provincia"]=="LIMA")&
    (df["Municipalidad"].str.contains("DISTRITAL"))
    ].pivot_table(
    index  =["Municipalidad"],
    columns=["Año"],
    values =[
        "Avance %",
        ],
    aggfunc={
        "Avance %": "mean",
        },
    # fill_value=0,
    margins=False,
    margins_name="Total",
    dropna=True,
    observed=True,
).copy()

# Eliminar el nivel 0
df_gph.columns = df_gph.columns.droplevel(0)

# Columna promedio
df_gph["Prom."] = df_gph.mean(axis=1)

# ordenar segun promedio
df_gph = df_gph.sort_values(by="Prom.", ascending=False)

# Transformar en porcentajes
df_gph = round(df_gph,1)

df_gph

Año                                                 2015  2016  2017  2018  \
Municipalidad                                                                
MUNICIPALIDAD DISTRITAL DE JESUS MARIA              80.7  85.4  85.6  90.7   
MUNICIPALIDAD DISTRITAL DE MIRAFLORES               94.0  92.7  95.7  94.9   
MUNICIPALIDAD DISTRITAL DE CARABAYLLO               92.8  86.9  89.0  71.5   
MUNICIPALIDAD DISTRITAL DE SAN MIGUEL               85.9  78.7  83.1  79.8   
MUNICIPALIDAD DISTRITAL DE MAGDALENA DEL MAR        82.5  94.4  77.8  71.3   
MUNICIPALIDAD DISTRITAL DE SAN BORJA                85.4  86.7  90.7  82.5   
MUNICIPALIDAD DISTRITAL DE SANTIAGO DE SURCO        85.9  82.0  82.3  86.3   
MUNICIPALIDAD DISTRITAL DE PUEBLO LIBRE             80.4  75.3  68.6  78.6   
MUNICIPALIDAD DISTRITAL DE LURIGANCHO (CHOSICA)     84.7  88.3  76.0  78.2   
MUNICIPALIDAD DISTRITAL DE LA MOLINA                92.0  85.1  82.3  70.8   
MUNICIPALIDAD DISTRITAL DE SAN MARTIN DE PORRES     84.3  86.8  75.2  87.7   
MUNICIPALIDAD DISTRITAL DE LURIN                    74.4  78.5  78.9  70.7   
MUNICIPALIDAD DISTRITAL DE LOS OLIVOS               90.9  71.4  72.3  79.2   
MUNICIPALIDAD DISTRITAL DE PUNTA HERMOSA            83.7  88.7  82.6  89.6   
MUNICIPALIDAD DISTRITAL DE SURQUILLO                73.5  82.3  84.4  84.5   
MUNICIPALIDAD DISTRITAL DE SANTA ROSA               77.8  84.3  87.3  84.7   
MUNICIPALIDAD DISTRITAL DE LINCE                    76.6  82.9  85.0  80.7   
MUNICIPALIDAD DISTRITAL DE ATE - VITARTE            84.0  71.3  71.5  82.9   
MUNICIPALIDAD DISTRITAL DE SAN ISIDRO               79.9  87.3  83.8  89.3   
MUNICIPALIDAD DISTRITAL DE COMAS                    85.7  84.7  69.7  68.3   
MUNICIPALIDAD DISTRITAL DE LA VICTORIA              80.7  70.8  68.1  70.6   
MUNICIPALIDAD DISTRITAL DE PUCUSANA                 85.1  75.0  85.1  84.6   
MUNICIPALIDAD DISTRITAL DE BARRANCO                 75.9  73.7  61.4  69.5   
MUNICIPALIDAD DISTRITAL DE SANTA MARIA DEL MAR      83.0  82.5  71.1  63.4   
MUNICIPALIDAD DISTRITAL DE PUENTE PIEDRA            76.1  72.7  75.8  76.0   
MUNICIPALIDAD DISTRITAL DE INDEPENDENCIA            85.9  85.9  72.6  75.1   
MUNICIPALIDAD DISTRITAL DE CIENEGUILLA              80.4  83.7  78.8  84.2   
MUNICIPALIDAD DISTRITAL DE SAN JUAN DE MIRAFLORES   71.5  81.4  72.2  62.9   
MUNICIPALIDAD DISTRITAL DE CHACLACAYO               69.8  79.1  63.6  71.5   
MUNICIPALIDAD DISTRITAL DE ANCON                    74.2  58.1  78.5  59.7   
MUNICIPALIDAD DISTRITAL DE SANTA ANITA              63.9  64.1  62.7  82.3   
MUNICIPALIDAD DISTRITAL DE PACHACAMAC               77.0  72.1  79.3  59.5   
MUNICIPALIDAD DISTRITAL DE SAN JUAN DE LURIGANCHO   73.6  80.9  84.8  84.4   
MUNICIPALIDAD DISTRITAL DE VILLA EL SALVADOR        87.7  81.8  77.1  71.1   
MUNICIPALIDAD DISTRITAL DE RIMAC                    91.4  81.1  83.0  69.2   
MUNICIPALIDAD DISTRITAL DE PUNTA NEGRA              87.6  79.9  77.2  74.4   
MUNICIPALIDAD DISTRITAL DE CHORRILLOS               83.7  78.4  75.0  78.8   
MUNICIPALIDAD DISTRITAL DE SAN BARTOLO              75.9  67.9  68.0  73.9   
MUNICIPALIDAD DISTRITAL DE BREÑA                    91.6  51.0  64.5  81.9   
MUNICIPALIDAD DISTRITAL DE SAN LUIS                 67.9  44.3  84.7  65.8   
MUNICIPALIDAD DISTRITAL DE EL AGUSTINO              62.1  47.6  75.3  69.1   
MUNICIPALIDAD DISTRITAL DE VILLA MARIA DEL TRIUNFO  83.8  77.6  66.6  69.6   

Año                                                 2019  2020  2021  2022  \
Municipalidad                                                                
MUNICIPALIDAD DISTRITAL DE JESUS MARIA              90.8  87.8  87.3  91.5   
MUNICIPALIDAD DISTRITAL DE MIRAFLORES               83.2  89.1  85.1  80.6   
MUNICIPALIDAD DISTRITAL DE CARABAYLLO               86.3  82.1  95.4  91.2   
MUNICIPALIDAD DISTRITAL DE SAN MIGUEL               85.5  80.2  85.1  92.3   
MUNICIPALIDAD DISTRITAL DE MAGDALENA DEL MAR        82.7  79.2  92.8  89.3   
MUNICIPALIDAD DISTRIT

En cuanto a la **ejecución presupuestal**, distritos como **Jesús María, Miraflores y Carabayllo** registraron los mayores porcentajes, superando el **86%** en promedio. En contraste, **San Luis, El Agustino y Villa María del Triunfo** presentaron niveles inferiores al **72%**, aunque han mostrado una tendencia de crecimiento en los últimos años.

![](../04_results/01_graphs/gph_07b.png)

### Ranking Distritos de Lima Metropolitana

In [56]:
# Grafico Ranking de Distritos de Lima Meretropolitana por PIM vs Avance V2

# Data para el gráfico
df_gph = df.loc[
    (df["Año"] == 2024)&
    (df["Provincia"]=="LIMA")&
    (df["Municipalidad"].str.contains("DISTRITAL"))
    ].pivot_table(
    index  =["Municipalidad"],
    # columns=["Año"],
    values =[
        "PIM",
        "Avance %",
        ],
    aggfunc={
        "PIM": lambda x: x.sum() / 1_000_000,
        "Avance %": "mean",
        },
    # fill_value=0,
    margins=False,
    margins_name="Total",
    dropna=True,
    observed=True,
).copy()

# ordenar segun promedio
df_gph = df_gph.sort_values(by="PIM", ascending=True).reset_index().round(1)

# Ranking
df_gph["Rank. PIM"] = df_gph["PIM"].rank(ascending=False, method="first").astype(int)
df_gph["Rank. Avance"] = df_gph["Avance %"].rank(ascending=False, method="first").astype(int)

df_gph

Municipalidad  Avance %    PIM  \
0      MUNICIPALIDAD DISTRITAL DE SANTA MARIA DEL MAR      93.0   10.7   
1              MUNICIPALIDAD DISTRITAL DE SAN BARTOLO      81.3   15.5   
2              MUNICIPALIDAD DISTRITAL DE PUNTA NEGRA      84.4   18.2   
3                 MUNICIPALIDAD DISTRITAL DE PUCUSANA      80.4   24.9   
4               MUNICIPALIDAD DISTRITAL DE SANTA ROSA      80.1   26.3   
5               MUNICIPALIDAD DISTRITAL DE CHACLACAYO      81.5   27.9   
6                    MUNICIPALIDAD DISTRITAL DE ANCON      89.0   32.8   
7                 MUNICIPALIDAD DISTRITAL DE SAN LUIS      87.5   34.4   
8              MUNICIPALIDAD DISTRITAL DE CIENEGUILLA      76.3   39.1   
9            MUNICIPALIDAD DISTRITAL DE PUNTA HERMOSA      81.8   46.0   
10                   MUNICIPALIDAD DISTRITAL DE BREÑA      82.0   46.0   
11                MUNICIPALIDAD DISTRITAL DE BARRANCO      81.9   61.4   
12                   MUNICIPALIDAD DISTRITAL DE LINCE      90.1   65.5   
13            MUNICIPALIDAD DISTRITAL DE PUEBLO LIBRE      93.3   66.5   
14               MUNICIPALIDAD DISTRITAL DE SURQUILLO      92.5   69.9   
15                   MUNICIPALIDAD DISTRITAL DE RIMAC      83.9   77.6   
16       MUNICIPALIDAD DISTRITAL DE MAGDALENA DEL MAR      92.9   81.3   
17              MUNICIPALIDAD DISTRITAL DE PACHACAMAC      72.1   85.0   
18             MUNICIPALIDAD DISTRITAL DE EL AGUSTINO      74.2   96.9   
19           MUNICIPALIDAD DISTRITAL DE INDEPENDENCIA      88.8   97.8   
20             MUNICIPALIDAD DISTRITAL DE SANTA ANITA      86.4   99.1   
21                   MUNICIPALIDAD DISTRITAL DE LURIN      91.8  106.4   
22             MUNICIPALIDAD DISTRITAL DE JESUS MARIA      94.3  107.3   
23              MUNICIPALIDAD DISTRITAL DE SAN MIGUEL      93.9  126.2   
24    MUNICIPALIDAD DISTRITAL DE LURIGANCHO (CHOSICA)      98.7  132.9   
25              MUNICIPALIDAD DISTRITAL DE LOS OLIVOS      92.9  141.0   
26              MUNICIPALIDAD DISTRITAL DE CARABAYLLO      85.8  143.8   
27               MUNICIPALIDAD DISTRITAL DE LA MOLINA      93.7  150.1   
28  MUNICIPALIDAD DISTRITAL DE SAN JUAN DE MIRAFLORES      85.9  154.1   
29             MUNICIPALIDAD DISTRITAL DE LA VICTORIA      97.1  163.0   
30               MUNICIPALIDAD DISTRITAL DE SAN BORJA      89.1  170.2   
31           MUNICIPALIDAD DISTRITAL DE PUENTE PIEDRA      86.8  179.5   
32                   MUNICIPALIDAD DISTRITAL DE COMAS      87.3  181.2   
33              MUNICIPALIDAD DISTRITAL DE CHORRILLOS      86.2  194.7   
34  MUNICIPALIDAD DISTRITAL DE VILLA MARIA DEL TRI...      71.1  202.4   
35       MUNICIPALIDAD DISTRITAL DE VILLA EL SALVADOR      85.2  203.6   
36    MUNICIPALIDAD DISTRITAL DE SAN MARTIN DE PORRES      81.1  250.8   
37           MUNICIPALIDAD DISTRITAL DE ATE - VITARTE      86.0  320.4   
38              MUNICIPALIDAD DISTRITAL DE SAN ISIDRO      72.8  339.6   
39              MUNICIPALIDAD DISTRITAL DE MIRAFLORES      90.9  370.7   
40       MUNICIPALIDAD DISTRITAL DE SANTIAGO DE SURCO      96.1  385.3   
41  MUNICIPALIDAD DISTRITAL DE SAN JUAN DE LURIGANCHO      88.2  446.6   

    Rank. PIM  Rank. Avance  
0          42             8  
1          41            34  
2          40            28  
3          39            36  
4          38            37  
5          37            33  
6          36            16  
7          35            19  
8          34            38  
9          32            32  
10         33            30  
11         31            31  
12         30            14  
13         29             7  
14         28            11  
15         27            29  
16         26             9  
17         25            41  
18         24            39  
19         23            17  
20         22            22  
21         21            12  
22         20             4  
23         19             5  
24         18             1  
25         17            10  
26         16            26  
27         15    

Finalmente, al rankear los distritos por PIM en **2024** y compararlos con su nivel de ejecución presupuestal, se observa que **Villa María del Triunfo, Villa El Salvador** y **San Isidro**, a pesar de estar en el **top 10** de los distritos con mayor asignación de PIM, se encuentran entre los **10 últimos** en el ranking de avance de ejecución.

Por otro lado, **Santa María del Mar, Magdalena del Mar** y **Pueblo Libre** lograron ubicarse en el **top 10** de ejecución presupuestal, a pesar de contar con una asignación de PIM inferior a la mitad de la de los distritos con mayor presupuesto.

![](../04_results/01_graphs/gph_08b.png)
<!-- <img src="../04_results/01_graphs/gph_08b.png" alt="Gráfico" width="600" height="auto"> -->


Estos resultados plantean interrogantes fundamentales sobre la gestión del presupuesto a nivel distrital.
* ¿Por qué algunos distritos con mayores asignaciones de PIM presentan bajos niveles de ejecución?.

* ¿Podría deberse a una menor **capacidad de absorción del gasto público** cuando enfrentan presupuestos significativamente mayores a los que están acostumbrados a manejar?.

* ¿**Mayores asignaciones** pueden dispersar su presupuesto en múltiples iniciativas, aumentando el **riesgo de retrasos o inejecución**?

* ¿Existe mas burocracia o deficiencias en la planificación?.

Por otro lado, los distritos con menor presupuesto han logrado niveles de ejecución más eficientes. 
* ¿Qué estrategias han implementado para optimizar sus recursos? 

* ¿Acaso la **urgencia en el gasto** los lleva a priorizar proyectos esenciales y ejecutarlos con mayor rapidez?

Una **alta ejecución** presupuestal no siempre es sinónimo de **eficiencia**, así como una **baja ejecución** no implica necesariamente **mala gestión**. Detrás de ambos escenarios pueden existir riesgos y desafíos. 

* Una rápida ejecución podría estar enfocada en proyectos de menor relevancia o responder a presiones por cumplir metas a corto plazo, ocultando deficiencias en la planificación y falta de supervisión. 

* Una baja ejecución podría evidenciar dificultades administrativas, limitaciones en la capacidad de gestión o retrasos en la implementación de proyectos de alto impacto.

**¿cómo evaluar si el gasto público se traduce en inversiones de alto impacto y beneficios reales para la población?**